massa-execution-worker/src/speculative_roll_state.rs

```diff
@@ -476,4 +507,56 @@ impl SpeculativeRollState {
 
         credits
     }
+
+    /// Remove a given amount from deferred credits and return the removed amount
+    pub fn remove_deferred_credits(&mut self, slot: &Slot, addr: &Address, amount: &Amount) -> Amount {
+
+        let amount_zero = Amount::zero();
+        let mut amount_to_rm = *amount;
+
+        // First try to rm in added_changes
+        let res = self.added_changes.deferred_credits.sub_amount(slot, addr, &amount_to_rm);
+        if let Some(a) = res {
+            amount_to_rm = amount_to_rm.saturating_sub(a);
+        }
+
+        // If not found, try to rm from active history
+        // Early test to avoid locking active history
+        if amount_to_rm > amount_zero {
+            let mut hist = self.active_history.write();
+            let mut hist_iter = hist.0.iter_mut().rev();
+            while amount_to_rm > amount_zero {
```

Comment:
This does not behave properly: we might "slash" some history elements that are overwritten later in history or through added_changes